### PR TITLE
[EOBS-196] Super User can not give EOBS users roles when creating them

### PR DESCRIPTION
**Pull request by moderatemisbehaviour**
Added conditions to allow operations when superuser is being used.

---

Before this pull request can be merged the following must be true.
- [ ] Unit tests pass (Travis integration).
- [ ] No code quality issues (Codacy integration).
- [ ] Approval from at least one developer and at least one tester.

If you are a BJSS contributor there are some additional conditions.
- [ ] All client module unit tests pass.
- [ ] The *Pull Request* field in the work tab of the JIRA issue is populated.
- [ ] The JIRA issue contains a description of the root cause and the solution in the comments.
- [ ] There are no bugs against the JIRA issues in the current sprint.

### CODE REVIEW

File: user_management/tests/nh_clinical_user_management/test_create.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        self.assertTrue('Employee' in groups,
+                        msg="User created without Employee group")
+
+    def test_shift_coordinator_can_create_hca(self):
+        """
+        Test that creating a user, creates a user
+        """
+        self.call_test(
+            created_user_role=self.hca_role,
+            user_to_create_with=self.superuser
```
Comment (Nayira) [inline user_management/tests/nh_clinical_user_management/test_create.py]: If the test is `test_shift_coordinator_can_create_hca`, shouldn't `user_to_create_with` be `shift_coordinator`?

File: user_management/tests/nh_clinical_user_management/test_create.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        )
+        self._assert_hca_created()
+
+    def test_superuser_can_create_hca(self):
+        self.call_test(
+            created_user_role=self.hca_role,
+            user_to_create_with=self.shift_coordinator
```
Comment (Nayira) [inline user_management/tests/nh_clinical_user_management/test_create.py]: As above, If the test is `test_superuser_can_create_hca`, shouldn't `user_to_create_with` be `superuser`?
Seems the user was swapped by mistake :)

File: user_management/tests/nh_clinical_user_management/test_write.py [hunk 3]
```diff
@@ -1,65 +1,89 @@
...
+        """
+        Test that the superuser can update subordinate user groups.
+        """
+        self.call_test()
+        update_data = {
+            'name': 'TU2',
+            'login': 'tu2',
+            'category_id': [[6, 0, [self.nurse_role.id]]]
+        }
+        update = self.hca.write(update_data)
```
Comment (Nayira) [inline user_management/tests/nh_clinical_user_management/test_write.py]: Could we make it so it updated it to be a 'System Admin'? That way we cover for the 1 clear need we have for the superuser setting roles :)

File: user_management/tests/nh_clinical_user_management/test_create.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        """
+        Test that creating a user, creates a user
+        """
+        self.call_test(
+            created_user_role=self.hca_role,
+            user_to_create_with=self.superuser
+        )
+        self._assert_hca_created()
+
+    def test_superuser_can_create_hca(self):
```
Comment (Nayira) [inline user_management/tests/nh_clinical_user_management/test_create.py]: Discussed with Dan - to add a test to create_user with role of Sys Admin

**Comment by moderatemisbehaviour** [comment]
@Nayira unit tests added in latest commit. Also had to merge develop in.

**Pull request merged**